lib/cartopy/_trace.cpp

```diff
@@ -558,7 +563,7 @@ void _project_segment(GEOSContextHandle_t handle,
     t_current = 0.0;
     state = get_state(p_current, gp_domain, handle);
 
-    while(t_current < 1.0)
+    while(t_current < 1.0 && lines.size() < 100)
```

Comment:
The number of _lines_ is usually small (< 10); the number of _points_ could be much larger (e.g., up to ~4000 per line.) Note: my numbers are based on the test suite; maybe something else might require more lines.

The scaling is roughly linear. For the example in #325, a limit of 100 is about 0.038 s, 1000 is 0.38 s 10000 is 3.9 s. For the plotting from #389, 1000 is a minor but perceptible slowdown, 10000 is heavily lagged.